components/builder-jobsrv/src/data_store.rs

```diff
@@ -154,13 +154,34 @@ impl DataStore {
                                 END
                                 $$ LANGUAGE plpgsql VOLATILE"#)?;
 
+        migrator.migrate("jobsrv",
+                         r#"CREATE OR REPLACE FUNCTION reset_jobs_v1 () RETURNS SETOF jobs AS
+                                $$
+                                DECLARE
+                                    r jobs % rowtype;
+                                BEGIN
+                                    FOR r IN
+                                        SELECT * FROM jobs
+                                        WHERE job_state = 'Dispatched'
+                                        ORDER BY created_at ASC
+                                        FOR UPDATE SKIP LOCKED
+                                    LOOP
+                                        UPDATE jobs SET job_state='Pending', updated_at=now() WHERE id=r.id RETURNING * INTO r;
+                                        RETURN NEXT r;
+                                    END LOOP;
+                                  RETURN;
+                                END
+                                $$ LANGUAGE plpgsql VOLATILE"#)?;
+
         // Update the state of a job. Takes a job id and a state, and updates that row.
         migrator.migrate("jobsrv",
                          r#"CREATE OR REPLACE FUNCTION set_job_state_v1 (jid bigint, jstate text) RETURNS void AS $$
                             BEGIN
                                 UPDATE jobs SET job_state=jstate, updated_at=now() WHERE id=jid;
                             END
                          $$ LANGUAGE plpgsql VOLATILE"#)?;
+
+        migrator.migrate("jobsrv", r#"DROP INDEX IF EXISTS pending_jobs_index_v1;"#)?;
```

Comment:
Are you suggesting removing the index entirely or the drop?  We need the drop since there is no 'create index if it doesn't already exist' command.